web/client/epics/catalog.js

```diff
@@ -174,24 +175,35 @@ export default (API) => ({
                                     }
                                 }
                                 if (!record) {
-                                    return text;
+                                    return [text];
                                 }
-                                return layer;
+                                return [layer, filter];
                             }));
                         }
                         return Rx.Observable.empty();
                     });
             })
             .mergeMap(results => {
                 if (results) {
-                    const allRecordsNotFound = results.filter(r => isString(r)).join(" ");
+                    const allRecordsNotFound = results.filter(r => isString(r[0])).join(" ");
                     let actions = [];
                     if (allRecordsNotFound) {
                         // return one notification for all records that have not been found
                         actions = [recordsNotFound(allRecordsNotFound)];
                     }
                     // add all layers found to the map
-                    actions = [...actions, ...results.filter(r => isObject(r)).map(r => addLayer(r))];
+                    actions = [
+                        ...actions,
+                        ...results.filter(r => isObject(r[0])).map(r => {
+                            if (r[1]) {
+                                r[0].params = {
+                                    ...r[0].params,
+                                    CQL_FILTER: r[1]
```

Comment:
just spread related params object here

in the action the params will be like
/#/viewer/openlayers/new?actions=[{"type":"CATALOG:ADD_LAYERS_FROM_CATALOGS","layers":["tiger_roads","tiger_roads","tiger_roads"],"sources":["gs_stable_wms","gs_stable_wms","gs_stable_wms"],"filters":[{CQL_FILTER: "NAME='5th Ave'"},{ CQL_FILTER: "NAME='Brooklyn Brg'"},{CQL_FILTER:"NAME='South End Ave'"}]}]

Reply:
Updated code to use lodash merge to merge default layer options with values passed by "options" object in listed action.